src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/AnalyzerConfigOptionsExtensions.cs

```diff
@@ -10,22 +11,34 @@
 
 namespace Microsoft.CodeAnalysis
 {
-    internal static partial class AnalyzerConfigOptionsExtensions
+    internal static class AnalyzerConfigOptionsExtensions
     {
-        public static T GetOption<T>(this AnalyzerConfigOptions optionSet, Option<T> option)
+#if CODE_STYLE
+        public static T GetOption<T>(this AnalyzerConfigOptions analyzerConfigOptions, PerLanguageOption<T> option, string language)
+        {
+            // Language is not used for .editorconfig lookups
+            _ = language;
+
+            return GetOption(analyzerConfigOptions, option);
+        }
+#endif
+
+        public static T GetOption<T>(this AnalyzerConfigOptions analyzerConfigOptions, Option<T> option)
         {
-            if (!TryGetEditorConfigOption(optionSet, option, out T value))
+            if (!TryGetEditorConfigOption(analyzerConfigOptions, option, out T value))
             {
+                Debug.Fail("Failed to find a .editorconfig key for the option.");
```

Comment:
I like this - this should prevent us from accidentally querying non-editorconfig based option from CodeStyle layer.